### DISCUSSION TITLE
NonlinearEigen executioner with ArrayVariable

### DISCUSSION THREAD

**Discussion by crb3874** [idaholab/moose#18935] [Q&A General] [answered]
Hello,

I am implementing a simple eigenvalue multi-group neutron diffusion application, using ArrayVariable/Kernels for condensing energy groups to avoid the need for verbose kernel definitions for scattering/fission/etc. in the input file.

I'd like to use the NonlinearEigen executioner for this, which requires deriving the fission kernel from the EigenKernel class. However, there does not appear to be an analagous "ArrayEigenKernel" to derive from for my specific case, as there are for ArrayKernels/ArrayDGKernels. 

Is there compatibility currently for doing this? If not, would the sensible path forward be to implement my own ArrayEigenKernel class?

Thanks

**Comment by GiudGiud** [comment]
@YaqiWang will be able to give you some clues here

**Comment by YaqiWang** [comment, marked as answer]
I suggest to work with `Eigenvalue` executioner with which the eigen kernels are derived from the same base as other kernels except are tagged differently in the input file.  The folder `moose/tests/problems/eigen_problem/arraykernels` contains few examples.

**Reply by crb3874** [reply]
Yaqi, thanks for the quick response. I'll take a look at these examples.
